intl_translation_format/README.md

```diff
@@ -56,6 +56,169 @@ pub run intl_translation_format:generate
 ```
 
 ### Implement a new translation file format
-[TBD]
 
+The intl_translation_format brings the tools needed to support a new translation format. 
+You can extend the class TranslationFormat and complete the functions that will take care of the extracting of the messages
+and create templates. There are several classes to make this task easier: `MonoLingualFormat`, `BinaryMonolingualFormat`,
+`MultiLingualFormat`.
 
+As an example, this guide will saw you how the JSON file format is implemented.
+
+1. Learning about your format
+
+It is important to do some research about the format you are interested to implement. Some good questions would be:
+ 1. Is it a standard? 
+ 2. Is it a binary or text-based format?
+ 3. Does it contain a translation for multiple languages per file or is it just one language per file?
+ 4. How does it handle Plurals, Genders, and Variables?
+
+Let's answer these questions for our new JSON format:
+ 1. While it is a standard data interchange format, there is no official standard for translations. Multiple frameworks create their implementations like `i18next` or `ng2-translate` for `Angular`.
+
+ Because the `ICU Format` is the core of the dart intl library, we decided to create a value/key simple JSON with ICU messages. 
+
+ As an example: 
+ ```json
+ {
+   "message1" : "This is a message",
+   "message2" : "Hello { userName }, this a message with a variable",
+   "message2" : "You have { howMany, plural, =0 { none Apples} one { one Apple} other { {howMany} Apples} }"
+ } 
+ ```
+ 2. It is a text-based format
+ 3. It contains one locale translation per file.
+ 4. We will use the ICU standard for plurals, genders, and variables.
+
+
+2. Start coding
+
+ - Create a new dart package and add intl_translation_format as a dependency.
+
+- Choose what `TranslationFormat` class to extend. In our case, our format will be monolingual and text-based so we will extend `MonolingualFormat`. You can see the implementation of other classes in other packages.
+
+```dart 
+class JsonFormat extends MonoLingualFormat {
+
+  @override
+  String get fileExtension => 'json';
+  ...
+}  
+```
+The first thing will be to add the `fileExtension` in our case `JSON`. You can override the `fileExtensions` list param in case your format supports multiple extensions (eg. Xliff supports .xliff and xlf).
+
+3. Implement the method to generate a template.
+
+This method will generate a file in the desired format that will contain the messages from the template.
```

Comment:
You should briefly explain, or at least link, an explanation of what a template is. Or maybe move the explanation of the parameter before the code.